common-arcana.lic

```diff
@@ -234,6 +234,34 @@ module DRCA
       .map { |_name, properties| properties['abbrev'] }
       .each { |abbrev| fput("release #{abbrev}") }
   end
+  
+  def parse_regalia  # generates an array of currently-worn regalia armor nouns
+    return unless DRStats.trader?
+    DRC.bput('inv armor', 'Type INVENTORY HELP for more options')
+    snapshot = reget(40)
+    if snapshot.grep(/All of your armor|You aren't wearing anything like that/).any? && snapshot.grep(/Type INVENTORY HELP/).any?
+      result = snapshot
+        .map(&:strip)
+        .reverse
+        .take_while { |item| !['All of your armor:', "You aren't wearing anything like that."].include?(item) }
+        .drop_while { |item| item != '[Type INVENTORY HELP for more options]' }
+        .drop(1)
+        .select { |item| item.include?('rough-cut crystal') || item.include?('faceted crystal') || item.include?('resplendent crystal')}
+    else
+      parse_regalia
```

Comment:
❓ Should this be `return parse_regalia`? As currently written, it looks like it will call `parse_regalia` but then exit the `else` and return the value of `result.each_...` on the next line.

Reply:
Yep, I think so.  I cribbed most of this from get_combat_items in equipmanager, though I wanted it to accomplish different things.  I see what I added later undermines that catch.

If I understand correctly I could change this easily when I'm making the other regalia changes, since it won't be changing the method declaration or what is returns.  No one will be inconvenienced if their common-arcana stays un-updated for a few days.

Reply:
👍